Python stereoCalibrate fails assertion when adding CALIB_USE_EXTRINSIC_GUESS flag

##### System information (version)
- OpenCV => 3.4.2
- Operating System / Platform => Raspbian Stretch
- Python => 3.5.3

##### Detailed description

When adding the CALIB_USE_EXTRINSIC_GUESS flag, stereoCalibrate will fail the CV_IS_MAT tests. When calling the same method without that flag, and with same arguments it completes without issue. Also calling stereoCalibrateExtended works both with and without the USE_EXTRINISC_GUESS flag. 

Not sure if this is related to #11085

##### Steps to reproduce

```.py
flags = cv2.CALIB_FIX_INTRINSIC
ret, mtx1, dist1, mtx2, dist2, R, T, E, F = cv2.stereoCalibrate(objpoints, imgpointsA, imgpointsB, cameraMatrix1, distCoeffs1, cameraMatrix2, distCoeffs2, grayA.shape[::-1], R=R, T=T, flags=flags, **kws)
flags += cv2.CALIB_USE_EXTRINSIC_GUESS
ret, mtx1, dist1, mtx2, dist2, R, T, E, F = cv2.stereoCalibrate(objpoints, imgpointsA, imgpointsB, cameraMatrix1, distCoeffs1, cameraMatrix2, distCoeffs2, grayA.shape[::-1], R=R, T=T, flags=flags, **kws)
```

Error output is:

> cv2.error: OpenCV(3.4.2) /home/pi/opencv-3.4.2/modules/calib3d/src/calibration.cpp:1781: error: (-215:Assertion failed) (((_imagePoints1) != __null && (((const CvMat*)(_imagePoints1))->type & 0xFFFF0000) == 0x42420000 && ((const CvMat*)(_imagePoints1))->cols > 0 && ((const CvMat*)(_imagePoints1))->rows > 0) && ((const CvMat*)(_imagePoints1))->data.ptr != __null) && (((_imagePoints2) != __null && (((const CvMat*)(_imagePoints2))->type & 0xFFFF0000) == 0x42420000 && ((const CvMat*)(_imagePoints2))->cols > 0 && ((const CvMat*)(_imagePoints2))->rows > 0) && ((const CvMat*)(_imagePoints2))->data.ptr != __null) && (((_objectPoints) != __null && (((const CvMat*)(_objectPoints))->type & 0xFFFF0000) == 0x42420000 && ((const CvMat*)(_objectPoints))->cols > 0 && ((const CvMat*)(_objectPoints))->rows > 0) && ((const CvMat*)(_objectPoints))->data.ptr != __null) && (((_npoints) != __null && (((const CvMat*)(_npoints))->type & 0xFFFF0000) == 0x42420000 && ((const CvMat*)(_npoints))->cols > 0 && ((const CvMat*)(_npoints))->rows > 0) && ((const CvMat*)(_npoints))->data.ptr != __null) && (((matR) != __null && (((const CvMat*)(matR))->type & 0xFFFF0000) == 0x42420000 && ((const CvMat*)(matR))->cols > 0 && ((const CvMat*)(matR))->rows > 0) && ((const CvMat*)(matR))->data.ptr != __null) && (((matT) !=__null && (((const CvMat*)(matT))->type & 0xFFFF0000) == 0x42420000 && ((const CvMat*)(matT))->cols > 0 && ((const CvMat*)(matT))->rows > 0) && ((const CvMat*)(matT))->data.ptr != __null) in function 'cvStereoCalibrateImpl'

```.py
# This one does not fail
flags = cv2.CALIB_FIX_INTRINSIC
flags += cv2.CALIB_USE_EXTRINSIC_GUESS
cv2.stereoCalibrateExtended(objpoints, imgpointsA, imgpointsB, cameraMatrix1, distCoeffs1, cameraMatrix2, distCoeffs2, grayA.shape[::-1], R, T, flags=flags)
```

Sorry it took a while to circle back to this.  It looks like matR is causing the problem:

`cv2.error: OpenCV(3.4.2) /home/pi/opencv-3.4.2/modules/calib3d/src/calibration.cpp:1783: error: (-215:Assertion failed) (((matR) != __null && (((const CvMat*)(matR))->type & 0xFFFF0000)== 0x42420000 && ((const CvMat*)(matR))->cols > 0 && ((const CvMat*)(matR))->rows > 0) && ((const CvMat*)(matR))->data.ptr != __null) in function 'cvStereoCalibrateImpl'`

Output of the new tests is:

```
======================================================================
FAIL: test_InputArray (test_misc.Arguments)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/opencv-3.4.2/modules/python/test/test_misc.py", line 52, in test_InputArray
    self.assertEqual(res1, "InputArray: empty()=true kind=0x00010000 flags=0x01010000 total(-1)=0 dims(-1)=0 size(-1)=0x0 type(-1)=CV_8UC1")
AssertionError: 'Inpu[23 chars]nd=0x100200010000 flags=0x01010000 total(-1)=0[37 chars]8UC1' != 'Inpu[23 chars]nd=0x00010000 flags=0x01010000 total(-1)=0 dim[33 chars]8UC1'
- InputArray: empty()=true kind=0x100200010000 flags=0x01010000 total(-1)=0 dims(-1)=0 size(-1)=0x0 type(-1)=CV_8UC1
?                                 ----
+ InputArray: empty()=true kind=0x00010000 flags=0x01010000 total(-1)=0 dims(-1)=0 size(-1)=0x0 type(-1)=CV_8UC1


======================================================================
FAIL: test_InputArrayOfArrays (test_misc.Arguments)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/opencv-3.4.2/modules/python/test/test_misc.py", line 70, in test_InputArrayOfArrays
    self.assertEqual(res1, "InputArrayOfArrays: empty()=true kind=0x00050000 flags=0x01050000 total(-1)=0 dims(-1)=1 size(-1)=0x0")
AssertionError: 'Inpu[31 chars]nd=0x100200050000 flags=0x01050000 total(-1)=0[20 chars]=0x0' != 'Inpu[31 chars]nd=0x00050000 flags=0x01050000 total(-1)=0 dim[16 chars]=0x0'
- InputArrayOfArrays: empty()=true kind=0x100200050000 flags=0x01050000 total(-1)=0 dims(-1)=1 size(-1)=0x0
?                                         ----
+ InputArrayOfArrays: empty()=true kind=0x00050000 flags=0x01050000 total(-1)=0 dims(-1)=1 size(-1)=0x0
```

--- a/modules/calib3d/src/calibration.cpp
+++ b/modules/calib3d/src/calibration.cpp
@@ -3541,7 +3541,7 @@ double cv::stereoCalibrate( InputArrayOfArrays _objectPoints,
                           TermCriteria criteria)
 {
     if (flags & CALIB_USE_EXTRINSIC_GUESS)
-        CV_Error(CV_StsBadFlag, "stereoCalibrate does not support CALIB_USE_EXTRINSIC_GUESS.");
+        CV_Error(Error::StsBadFlag, "stereoCalibrate does not support CALIB_USE_EXTRINSIC_GUESS.");
 
     Mat Rmat, Tmat;
     double ret = stereoCalibrate(_objectPoints, _imagePoints1, _imagePoints2, _cameraMatrix1, _distCoeffs1,

--- a/modules/calib3d/src/calibration.cpp
+++ b/modules/calib3d/src/calibration.cpp
@@ -3540,6 +3540,9 @@ double cv::stereoCalibrate( InputArrayOfArrays _objectPoints,
                           OutputArray _Emat, OutputArray _Fmat, int flags,
                           TermCriteria criteria)
 {
+    if (flags & CALIB_USE_EXTRINSIC_GUESS)
+        CV_Error(CV_StsBadFlag, "stereoCalibrate does not support CALIB_USE_EXTRINSIC_GUESS.");
+
     Mat Rmat, Tmat;
     double ret = stereoCalibrate(_objectPoints, _imagePoints1, _imagePoints2, _cameraMatrix1, _distCoeffs1,
                                  _cameraMatrix2, _distCoeffs2, imageSize, Rmat, Tmat, _Emat, _Fmat,

